dali/operators/random/rng_base_cpu.h

```diff
@@ -89,45 +126,76 @@ void RNGBase<Backend, Impl, IsNoiseGen>::RunImplTyped(workspace_t<CPUBackend> &w
     output.SetLayout(input.GetLayout());
   }
 
+  // TODO(janton): set layout explicitly from the user for RNG
+
   auto &dists_cpu = backend_data_.dists_cpu_;
   dists_cpu.resize(sizeof(Dist) * nsamples);  // memory was already reserved in the constructor
   Dist* dists = reinterpret_cast<Dist*>(dists_cpu.data());
   bool use_default_dist = !This().template SetupDists<T>(dists, nsamples);
 
+  int channel_dim = -1;
+  auto layout = output.GetLayout();
+  channel_dim = layout.empty() ? ndim - 1 : layout.find('C');
+
   DistGen<IsNoiseGen> dist_gen_;
   for (int sample_id = 0; sample_id < nsamples; ++sample_id) {
     auto sample_sz = out_shape.tensor_size(sample_id);
+    int64_t N = sample_sz;
```

Comment:
Can you rename N to something more meaningful?

Reply:
Will do